src/twitch/twitch-eventsub-stream.ts

```diff
@@ -1,101 +1,54 @@
 import { TwitchEventSubClient } from "./twitch-eventsub";
-import { Stream } from "../effect-exports";
 
-export type IChatMessageStream = Stream<CurrentlyPlayingRequestMessage>;
-
-const makeChatMessageStream = Effect.gen(function* () {
+const make = Effect.gen(function* () {
   const config = yield* TwitchConfig;
   const eventsub = yield* TwitchEventSubClient;
+  const pubsub = yield* MessagePubSub;
 
-  return Stream.async<CurrentlyPlayingRequestMessage>((emit) => {
-    eventsub.onChannelChatMessage(
+  yield* Effect.acquireRelease(
+    Effect.sync(() => eventsub.onChannelChatMessage(
       config.broadcasterId,
       config.broadcasterId,
-      async (event) => {
+      (event) => {
         if (event.messageText === "!song") {
-          await emit.single(
+          pubsub.unsafePublish(
             Message.CurrentlyPlayingRequest({
               requesterDisplayName: event.chatterDisplayName,
             }),
           );
         }
-      },
-    );
-  });
-});
-
-export class ChatMessageStream extends Context.Tag("chat-message-stream")<
-  ChatMessageStream,
-  IChatMessageStream
->() {
-  static Live = Layer.effect(this, makeChatMessageStream);
-}
-
-export type ISongRequestMessageStream = Stream<SongRequestMessage>;
-
-const makeSongRequestMessageStream = Effect.gen(function* () {
-  const config = yield* TwitchConfig;
-  const eventsub = yield* TwitchEventSubClient;
+      })),
+    sub => Effect.sync(() => sub.stop())
+  )
 
-  return Stream.async<SongRequestMessage>((emit) => {
-    eventsub.onChannelRedemptionAddForReward(
+  yield* Effect.acquireRelease(
+    Effect.sync(() => eventsub.onChannelRedemptionAddForReward(
       config.broadcasterId,
       config.songRequestRewardId,
       (event) =>
-        emit.single(
+        pubsub.unsafePublish(
           Message.SongRequest({
             requesterDisplayName: event.userDisplayName,
             url: event.input,
           }),
-        ),
-    );
-  });
+        )
+    )),
+    sub => Effect.sync(() => sub.stop())
+  )
 });
 
-export class SongRequestMessageStream extends Context.Tag(
-  "song-request-message-stream",
-)<SongRequestMessageStream, ISongRequestMessageStream>() {
-  static Live = Layer.effect(this, makeSongRequestMessageStream);
-}
-
-const make = Effect.gen(function* () {
-  const chatMessageStream = yield* ChatMessageStream;
-  const songRequestMessageStream = yield* SongRequestMessageStream;
-  const pubsub = yield* MessagePubSub;
-
-  const eventSubStream = Stream.merge(
-    chatMessageStream,
-    songRequestMessageStream,
-  );
-
-  yield* Effect.forkScoped(
-    Stream.runForEach(eventSubStream, (message) =>
-      Effect.gen(function* () {
-        yield* Effect.logInfo(
-          `Received ${message._tag} message from twitch eventsub from @${message.requesterDisplayName}`,
-        );
-
-        yield* pubsub.publish(message);
-      }),
-    ),
-  );
-});
+export const TwitchEventSubStreamLive = Layer.scopedDiscard(make).pipe(
```

Comment:
This service doesn't actually provide anything, so we can use one of the layer discard apis.